doc/userguide/src/ExtendingRobotFramework/CreatingTestLibraries.rst

```diff
@@ -1731,6 +1761,34 @@ Dynamic libraries must always have this method. If it is missing, or
 if calling it fails for some reason, the library is considered a
 static library.
 
+Marking methods to expose as keywords
+'''''''''''''''''''''''''''''''''''''
+
+If a dynamic library should contain both methods which are meant to be keywords
+and methods which are meant to be private helper methods, it may be wise to
+mark the keyword methods as such so it is easier to implement `get_keyword_names`.
+The `robot.api.deco.robot_name` decorator allows an easy way to do this since it
+creates an attribute on the decorated method which is not normally there (`robot_name`).
+This allows generating the list of keywords just by checking for the `robot_name`
+attribute on every method in the library during `get_keyword_names`.  See
+`Using a custom keyword name`_ for more about this decorator.
+
+.. sourcecode:: python
+
+   from robot.api.deco import robot_name
+
+   class DynamicExample:
+
+       def get_keyword_names(self):
+           return [name for name in self if callable(getattr(self, name)) and hasattr(self, name)]
```

Comment:
This looks strange. I don't see how `for name in self` would work unless you have a custom `__iter__`, and if it returns attributes, `hasattr(self, name)` ought to always be True.

Wouldn't something like this (not tested!) work better as a generic example:

```python
    return [name for name in dir(self) if hasattr(getattr(self, name), 'robot_name')]
```